.github/workflows/msys2-auto.yml

```diff
@@ -0,0 +1,55 @@
+name: msys2 auto
+
+on:
+  workflow_dispatch:
+  push:
+  pull_request:
+    branches: [ master ]
+    paths-ignore:
+      - '.github/CODEOWNERS'
+      - 'doc/**'
+      - 'release_docs/**'
+      - 'ACKNOWLEDGEMENTS'
+      - 'COPYING**'
+      - '**.md'
+
+permissions:
+  contents: read
+
+# Using concurrency to cancel any in-progress job or run
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref && github.ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        sys:
+          - mingw64
+          - mingw32
+          - ucrt64
+          - clang64
+    runs-on: windows-latest
+    steps:
+      - name: Setup MSYS2
+        uses: msys2/setup-msys2@v2
+        with:
+          msystem: ${{matrix.sys}}
+          update: true
+          install: >-
+            git
+            autotools
+          pacboy: >-
+            toolchain:p
+            libjpeg-turbo:p
+      - name: Test HDF4
+        shell: msys2 {0}
+        run: |
+          uname -a
+          git clone https://github.com/HDFGroup/hdf4.git
+          cd hdf4
+          ./autogen.sh
+          ./configure LDFLAGS="-lws2_32" --disable-netcdf --disable-netcdf-tools --enable-static-exec
+          make
```

Comment:
Please separate into steps - see any of the other *-auto.yml files.  Consistency.